code/modules/reagents/reagent_containers.dm

```diff
@@ -49,11 +53,34 @@
 	if(!possible_transfer_amounts)
 		src.verbs -= /obj/item/reagent_containers/verb/set_APTFT
 	create_reagents(volume)
+	shatter_material = new
 
-/obj/item/reagent_containers/attack_self(mob/user as mob)
+/obj/item/reagent_containers/attack_self(mob/user)
 	return
 
-/obj/item/reagent_containers/attack(mob/M as mob, mob/user as mob, def_zone)
+/obj/item/reagent_containers/throw_impact(atom/hit_atom, var/speed)
+	. = ..()
+	if((speed >= fragile) && !no_shatter)
+		shatter()
+
+/obj/item/reagent_containers/proc/shatter(var/obj/item/W, var/mob/user)
```

Comment:
What is W doing here? I don't see any children that use it and it isn't used in this proc itself.

Reply:
it's for if somebody wants to do something specific to the weapon a reagent container was shattered with in the future, e.g coating it